src/sample/models/Bot.java

```diff
@@ -16,24 +16,23 @@ public Bot(int movementBoundary)
         this.movementBoundary = movementBoundary;
     }
 
-    // made the interface instead to require a moveEnum move
-    // and a no argument moveRandomly
     public void Move(MoveEnum direction)
     {
         // Determine which moves are possible, then add them to the list
-        var possibleMoves = getPossibleMoves();
+        possibleMoves = getPossibleMoves();
         if (possibleMoves.contains(direction))
         {
             executeMove(direction);
         }
+        possibleMoves.clear();
     }
 
     // random bot movement
     public void MoveRandomly()
     {
         // Determine which moves are possible, then add them to the list
-        ArrayList<MoveEnum> possibleMoves = getPossibleMoves();
-        getPreferredMoves();
+        possibleMoves = getPossibleMoves();
+        preferredMoves = getPreferredMoves();
         Random randomNumberGenerator = new Random(System.currentTimeMillis());
```

Comment:
Better keep randomNumberGenerator  as a private field populated in the constructor (or even as a static).

Reply:
good point. fixed that and committed.